### PR TITLE
Make the AzureServiceTokenProvider mockable

### PR DESCRIPTION
**Pull request by christothes**
Make the AzureServiceTokenProvider mockable as requested in #7603

Mock example
```csharp
//bogus
var jwt = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c";
var mockProvider =  new Mock<AzureServiceTokenProvider>(() => new AzureServiceTokenProvider(null, "https://login.microsoftonline.com/"));
var mockTokenCallback = new Mock<AzureServiceTokenProvider.TokenCallback>();
mockProvider
    .SetupGet(p => p.KeyVaultTokenCallback)
    .Returns(mockTokenCallback.Object);
	
var kv = new KeyVaultClient(new KeyVaultClient.AuthenticationCallback(mockProvider.Object.KeyVaultTokenCallback));
mockTokenCallback
    .Setup(tc => tc(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<string>()))
    .ReturnsAsync (() => jwt);

try
{
    await kv.GetSecretAsync("https://someKeyVaultName.vault.azure.net", "someSecretName");
}
catch(KeyVaultErrorException ex)
{
    //Operation returned an invalid status code 'Unauthorized' (for bogus jwt)
    Console.WriteLine(ex.Message);
}
	
mockTokenCallback
    .Setup(tc => tc(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<string>()))
    .Throws(new Exception("foo"));

try
{
    await kv.GetSecretAsync("https://someKeyVaultName.vault.azure.net", "someSecretName");
}
catch (Exception ex)
{
    //foo
    Console.WriteLine(ex.Message);
}
```

### CODE REVIEW

File: sdk/mgmtcommon/AppAuthentication/Azure.Services.AppAuthentication/AzureServiceTokenProvider.cs [hunk 1]
```diff
@@ -44,7 +44,7 @@ public class AzureServiceTokenProvider
         /// KeyVaultClient keyVaultClient = new KeyVaultClient(new KeyVaultClient.AuthenticationCallback(azureServiceTokenProvider.KeyVaultTokenCallback));
         /// </code>
         /// </example>
-        public TokenCallback KeyVaultTokenCallback => async (authority, resource, scope) =>
+        public virtual TokenCallback KeyVaultTokenCallback => async (authority, resource, scope) =>
```
Comment (jongio) [inline sdk/mgmtcommon/AppAuthentication/Azure.Services.AppAuthentication/AzureServiceTokenProvider.cs]: Have we decided to use virtual vs interface to support mocks?  Tradeoffs either way, just want to be sure virtual is consistent across the rest of the SDK.

**Comment by jongio** [comment]
Please consider adding tests to ensure we don't break mockability in the future.

File: sdk/mgmtcommon/AppAuthentication/Azure.Services.AppAuthentication/AzureServiceTokenProvider.cs (hunk 1 above)
Comment (christothes) [reply sdk/mgmtcommon/AppAuthentication/Azure.Services.AppAuthentication/AzureServiceTokenProvider.cs]: virtual methods seem to be the standard adopted by most of the azure SDK components.

**Comment by pakrym** [comment]
Please be advised that this is considered to be a minor but breaking change.

**Comment by AlexGhiondea** [comment]
For breaking changes please ensure that you rev the package version.

**Comment by christothes** [comment]
> For breaking changes please ensure that you rev the package version.

Done

File: sdk/mgmtcommon/AppAuthentication/Azure.Services.AppAuthentication/Microsoft.Azure.Services.AppAuthentication.csproj [hunk 2]
```diff
@@ -2,7 +2,7 @@
     <PropertyGroup>
         <PackageId>Microsoft.Azure.Services.AppAuthentication</PackageId>
         <Description>Enables a service to authenticate to Azure services using the developer's Azure Active Directory/ Microsoft account during development, and authenticate as itself (using OAuth 2.0 Client Credentials flow) when deployed to Azure.</Description>
-        <Version>1.3.1</Version>
+        <Version>1.3.2</Version>
```
Comment (AlexGhiondea) [inline sdk/mgmtcommon/AppAuthentication/Azure.Services.AppAuthentication/Microsoft.Azure.Services.AppAuthentication.csproj]: I think you should bump minor. Packages roll forward on patch numbers

Comment (christothes) [reply sdk/mgmtcommon/AppAuthentication/Azure.Services.AppAuthentication/Microsoft.Azure.Services.AppAuthentication.csproj]: done

Comment (christothes) [reply sdk/mgmtcommon/AppAuthentication/Azure.Services.AppAuthentication/Microsoft.Azure.Services.AppAuthentication.csproj]: @AlexGhiondea could you take another look?

**Review by AlexGhiondea — APPROVED** [review]

**Pull request merged**